src/HigherOrderComponent/MapProvider/MapProvider.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import OlMap from 'ol/map';
+
+/**
+ * The MapProvider.
+ *
+ * @type {Object}
+ */
+class MapProvider extends React.Component {
+
+  /**
+   * The properties.
+   * @type {Object}
+   */
+  static propTypes = {
+    /**
+     * The children of this group. Typically a set of `ToggleButton`s.
+     * @type {Object}
+     */
+    children: PropTypes.node,
+
+    /**
+     * The map can be either an OlMap or a Promise that resolves with an OlMap
+     * if your map is created asynchronusly.
+     *
+     * @type {ol.Map|Promise}
+     */
+    map: PropTypes.oneOfType([
+      PropTypes.instanceOf(OlMap),
+      PropTypes.func
```

Comment:
I assume there is no way of actually telling apart any func from a true Promise, right?

Reply:
Maybe if we fix your following comment.